avni-server-api/src/main/java/org/avni/server/web/MetabaseController.java

```diff
@@ -0,0 +1,243 @@
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("x-api-key", apiKey);
+
+        Map<String, Object> details = new HashMap<>();
+        details.put("host", "localhost");
+        details.put("port", "5432");  
+        details.put("db", "openchs");  
+        details.put("user", dbUser);
+
+        Map<String, Object> requestBody = new HashMap<>();
+        requestBody.put("engine", "postgres");
+        requestBody.put("name", dbUser);  
+        requestBody.put("details", details);
+
+        HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
+        logger.info("Sending request to create database for user: {}", dbUser);
+        ResponseEntity<Map> response = restTemplate.exchange(url, HttpMethod.POST, entity, Map.class);
+        logger.info("Response from Metabase API (create database): " + response.getBody());
+
+        return (Integer) response.getBody().get("id");
+    }
+
+    private int createCollection(String name) {
+        String url = "http://localhost:3000/api/collection";
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("x-api-key", apiKey);
+
+        Map<String, Object> requestBody = new HashMap<>();
+        requestBody.put("name", name);
+        requestBody.put("description", name + " collection");
+
+        HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
+        logger.info("Sending request to create collection for name: {}", name);
+        ResponseEntity<Map> response = restTemplate.exchange(url, HttpMethod.POST, entity, Map.class);
+        logger.info("Response from Metabase API (create collection): " + response.getBody());
+
+        return (Integer) response.getBody().get("id");
+    }
+
+    private int createPermissionsGroup(String name) {
+        String url = "http://localhost:3000/api/permissions/group";
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("x-api-key", apiKey);
+
+        Map<String, Object> requestBody = new HashMap<>();
+        requestBody.put("name", name);
+
+        HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
+        logger.info("Request body for creating permissions group: {}", requestBody);
+        logger.info("Sending request to create permissions group for name: {}", name);
+
+        try {
+            ResponseEntity<Map> response = restTemplate.exchange(url, HttpMethod.POST, entity, Map.class);
+            logger.info("Response from Metabase API (create permissions group): " + response.getBody());
+            return (Integer) response.getBody().get("id");
+        } catch (Exception e) {
+            logger.error("Error creating permissions group for name: " + name, e);
+            throw new RuntimeException("Failed to create permissions group for name: " + name, e);
+        }
```

Comment:
Can you check what happens if you remove this try-catch block and if there is an error in the exchange? Take a look at https://github.com/avniproject/avni-server/blob/master/avni-server-api/src/main/java/org/avni/server/web/ErrorInterceptors.java

Reply:
Yes it throws an error indeed!